packages/roosterjs-editor-dom/lib/selection/Position.ts

```diff
@@ -131,46 +131,51 @@ export default class Position {
      * Get bounding rect of this position
      */
     getRect(): Rect {
-        let document = this.node && this.node.ownerDocument;
+        const document = this.node && this.node.ownerDocument;
 
         if (!document) {
             return null;
         }
 
-        let range = document.createRange();
+        const range = document.createRange();
         range.setStart(this.node, this.offset);
 
         // 1) try to get rect using range.getBoundingClientRect()
-        let rect = normalizeRect(range.getBoundingClientRect());
-
-        if (!rect) {
-            let { node, element, offset } = this.normalize();
-
-            // 2) if current cursor is inside text node, use range.getClientRects() for safari
-            // or insert a SPAN and get the rect of SPAN for others
-            if (Browser.isSafari) {
-                let rects = range.getClientRects();
-                if (rects && rects.length == 1) {
-                    rect = normalizeRect(rects[0]);
-                }
-            } else {
-                if (node.nodeType == NodeType.Text) {
-                    let span = document.createElement('SPAN');
-                    range.setStart(node, offset);
-                    range.collapse(true /*toStart*/);
-                    range.insertNode(span);
-                    rect = normalizeRect(span.getBoundingClientRect());
-                    span.parentNode.removeChild(span);
-                }
+        let rect = range.getBoundingClientRect ? normalizeRect(range.getBoundingClientRect()) : null;
+        if (rect) {
+            return rect;
+        }
+
+        // 2) try to get rect using range.getClientRects
+        const { node, element, offset } = this.normalize();
+        const rects = range.getClientRects ? Array.prototype.slice.call(range.getClientRects()) : [];
```

Comment:
Why this construction of `Array.prototype.slice.call`? Why grab a shallow copy if all your'e going to do with it is check it's length and then grab the first element?

Reply:
The goal was to satisfy type checking when creating the array to default to (type returned by getClientRects is not instantiable). But it's not needed I agree. Removed and defaulted to null. I already check if rects exists below.

Reply:
You could type the rects returned as appropriate for a type safeguard. like `const rects: T = blah`. But, this should be fine too.